src/test/java/org/wildfly/security/auth/client/MatchHostRuleTest.java

```diff
@@ -0,0 +1,120 @@
+package org.wildfly.security.auth.client;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.regex.Pattern;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+/**
+ * @author Tomas Hofman (thofman@redhat.com)
+ */
+public class MatchHostRuleTest {
+
+    @Test
+    public void testDomainName() throws URISyntaxException {
+        Assert.assertTrue(MatchRule.ALL.matchHost("security.wildfly.org").matches(new URI("remote+http://security.wildfly.org:9990")));
+        Assert.assertTrue(MatchRule.ALL.matchHost("security.wildfly.").matches(new URI("remote+http://security.wildfly.org:9990")));
+        Assert.assertTrue(MatchRule.ALL.matchHost("security.wildfly").matches(new URI("remote+http://security.wildfly.org:9990")));
+
+        Assert.assertFalse(MatchRule.ALL.matchHost("wildfly.org").matches(new URI("remote+http://security.wildfly.org:9990")));
+    }
+
+    @Test
+    public void testIPv4() throws URISyntaxException {
+        Assert.assertTrue(MatchRule.ALL.matchHost("127.0.0.1").matches(new URI("remote+http://127.0.0.1:9990")));
+        Assert.assertTrue(MatchRule.ALL.matchHost("127.0.0.").matches(new URI("remote+http://127.0.0.1:9990")));
```

Comment:
I think we definitely don't want this behavior; a string which is an invalid IP and an invalid host name should get kicked out, if possible.